api/v1alpha1/nodepool_types.go

```diff
@@ -13,6 +13,7 @@ const (
 	NodePoolValidAMIConditionType                = "ValidAMI"
 	NodePoolValidPowerVSImageConditionType       = "ValidPowerVSImage"
 	NodePoolValidMachineConfigConditionType      = "ValidMachineConfig"
+	NodePoolValidKubevirtConfigConditionType     = "ValidKubevirtConfig"
```

Comment:
To be consistent for consumers we should probably have a generic condition e.g. ValidProviderConfiguration where InvalidKubevirtConfig is the reason but will figure in follow up.

Reply:
Makes sense. I'd like to see a nodepool validation webhook similar to what we have for hostedcluster as well. I don't mind following up with this after this PR goes in. it probably does belong in another PR though since it will impact multiple platforms